Resources/Prototypes/_RMC14/Catalog/Fills/Boxes/attachment.yml

```
@@ -0,0 +1,48 @@
+- type: entity
+  name: Attachment Package
+  parent: BaseStorageItem
+  id: RMCBoxAttachment
+  description: A package containing various attachments
+  components:
+  - type: StorageFill
+    contents:
+    - id: RMCM5Bayonet
+      amount: 1
+    - id: RMCAttachmentMagneticHarness
+      amount: 1
+    - id: RMCAttachmentLaserSight
+      amount: 1
+  - type: Sprite
+    state: circuit
+    sprite: _RMC14/Objects/Storage/boxes.rsi
+  - type: Item
+    sprite: _RMC14/Objects/Storage/boxes.rsi
+    size: Tiny
+  - type: Storage
+    maxItemSize: Small
+    grid:
+    - 0,0,3,1
+    whitelist:
+      tags:
+      - RMCM5Bayonet
+      - RMCAttachmentMagneticHarness
+      - RMCAttachmentLaserSight
+  - type: InteractedBlacklist
+    blacklist:
+      components:
+      - Xeno
+  - type: Tag
+    tags:
+    - RMCBoxAttachment
+
+- type: Tag
+  id: RMCBoxAttachment
+
+# TODO RMC14: how this works, at least from my understanding and tests, is that it pulls attachments from this list and populates the box randomly. I ain't doing that for now, but good luck to however does it.
+#	var/list/common = list(/obj/item/attachable/suppressor, /obj/item/attachable/bayonet, /obj/item/attachable/flashlight)
+#	var/list/attachment_1 = list(/obj/item/attachable/reddot, /obj/item/attachable/burstfire_assembly, /obj/item/attachable/lasersight,
+#								/obj/item/attachable/extended_barrel,/obj/item/attachable/verticalgrip, /obj/item/attachable/angledgrip,
+#								/obj/item/attachable/gyro, /obj/item/attachable/bipod)
+#	var/list/attachment_2 = list(/obj/item/attachable/stock/smg, /obj/item/attachable/stock/shotgun, /obj/item/attachable/stock/rifle, /obj/item/attachable/magnetic_harness,
+#								/obj/item/attachable/heavy_barrel, /obj/item/attachable/scope,
+#								/obj/item/attachable/scope/mini)
```

SS13 code is under a license that's incompatible with ours, so just to be safe we shouldn't be including snippets of it
As for the random part itself, there are a couple components to do that later, like StorageFill
```suggestion
- type: entity
  name: Attachment Package
  parent: BaseStorageItem
  id: RMCBoxAttachment
  description: A package containing various attachments
  components:
  - type: StorageFill
    contents:
    - id: RMCM5Bayonet
      amount: 1
    - id: RMCAttachmentMagneticHarness
      amount: 1
    - id: RMCAttachmentLaserSight
      amount: 1
  - type: Sprite
    state: circuit
    sprite: _RMC14/Objects/Storage/boxes.rsi
  - type: Item
    sprite: _RMC14/Objects/Storage/boxes.rsi
    size: Tiny
  - type: Storage
    maxItemSize: Small
    grid:
    - 0,0,3,1
    whitelist:
      tags:
      - RMCM5Bayonet
      - RMCAttachmentMagneticHarness
      - RMCAttachmentLaserSight
  - type: InteractedBlacklist
    blacklist:
      components:
      - Xeno
  - type: Tag
    tags:
    - RMCBoxAttachment

- type: Tag
  id: RMCBoxAttachment

```